src/main/java/org/folio/entlinks/service/links/InstanceAuthorityLinkingService.java

```diff
@@ -181,4 +184,58 @@ private Specification<InstanceAuthorityLink> getSpecFromStatusAndDates(
       return builder.and(predicates.toArray(new Predicate[0]));
     };
   }
+
+  private void updateExistingLinks(List<InstanceAuthorityLink> incomingValidLinks,
+                                   List<InstanceAuthorityLink> existedLinks,
+                                   Map<UUID, AuthorityData> savedAuthorityData) {
+    for (InstanceAuthorityLink incomingLink : incomingValidLinks) {
+      AuthorityData linkAuthorityData = incomingLink.getAuthorityData();
+      AuthorityData authorityData = savedAuthorityData.get(linkAuthorityData.getId());
+      incomingLink.setAuthorityData(authorityData);
+      existedLinks.stream()
+        .filter(existedLink -> existedLink.isSameLink(incomingLink))
+        .findFirst()
+        .ifPresent(existedLink -> incomingLink.setId(existedLink.getId()));
+    }
+  }
+
+  private void updateLinkingRules(List<InstanceAuthorityLink> incomingLinks) {
+    var linkingRules = linkingRulesService.getLinkingRules().stream()
+      .collect(Collectors.toMap(InstanceAuthorityLinkingRule::getId, Function.identity()));
+
+    incomingLinks.forEach(link -> link.setLinkingRule(linkingRules.get(link.getLinkingRule().getId())));
+  }
```

Comment:
Why is it needed to set full rule object to links? It should be enough to have only the Id of the link reference to successfully save links to db.

Reply:
We use full rule objects during validation and constructing events.
So we can pass map of rules to each of this methods instead of filling links with rules. 
But it will look not so good because of utils overloaded methods and mapOfRules as parameter

![image](https://github.com/folio-org/mod-entities-links/assets/88006020/fc70e8ad-4093-4828-8623-fcbe939bdc72)